IPython/core/interactiveshell.py

```diff
@@ -440,6 +431,15 @@ def __init__(self, ipython_dir=None, profile_dir=None,
 
         # This is where traits with a config_key argument are updated
         # from the values on config.
+        cfg = kwargs.get('config', {})
+        parent_config = {}
+        parent = kwargs.get('parent', {})
+        if parent:
+            parent_config = getattr(parent, 'config', {})
```

Comment:
Do we actually need to look up parent config? Don't we pass the config in on instantiation, like `(..., parent=self, config=self.config)`?